src/app/server/Mdns.cpp

```diff
@@ -101,6 +103,201 @@ uint16_t GetUnsecuredPort()
     return gUnsecuredPort;
 }
 
+#define TIMEOUT_CLEARED 0
+Time::TimeSource<Time::Source::kSystem> gTimeSource;
+
+int16_t gDiscoveryTimeoutSecs   = CHIP_DEVICE_CONFIG_DISCOVERY_TIMEOUT_SECS;
+uint64_t gDiscoveryExpirationMs = TIMEOUT_CLEARED;
```

Comment:
Can't we at least fold of these globals into the `chip::Mdns::ServiceAdvertiser::Instance()` singleton?

Reply:
So I made chip::app::Mdns a singleton chip::app::MdnsServer.Instance() so that it can hold current app server state (which did not seem to easily belong in ServiceAdvertiser()).